aim/cli/up/commands.py

```diff
@@ -115,6 +118,76 @@ def up(dev, host, port, workers, repo, tf_logs, ssl_keyfile, ssl_certfile, base_
     click.echo('Press Ctrl+C to exit')
 
     try:
+        if tf_logs:
+            tf_logs_cache_path = f'{repo_inst.path}/tf_logs_cache.json'
+            try:
+                with open(tf_logs_cache_path) as FS:
+                    tf_logs_cache = json.load(FS)
+            except Exception:
+                tf_logs_cache = {}
+
+            for run_path in os.listdir(tf_logs):
+                path = os.path.join(tf_logs, run_path)
+                run_hash = tf_logs_cache.get(path)
+                if run_hash:
+                    run = Run(run_hash=run_hash)
+                    run_tf_hashsums = run['tf_hashsums']
+                else:
+                    run_tf_hashsums = {}
+                    run = Run()
+                    run['tf_hashsums'] = run_tf_hashsums
+                    tf_logs_cache[path] = run.hash
+
+                events = [
+                    os.path.join(root, file)
+                    for root, dirs, files in os.walk(path)
+                    for file in files if file.endswith('v2')
+                ]
+
+                events_to_process = []
+                for event in events:
+                    # Calculate file hash
+                    chunk_size = 8192
+                    file_hash = hashlib.md5()
+                    with open(event, 'rb') as FS:
+                        chunk = FS.read(chunk_size)
+                        while chunk:
+                            file_hash.update(chunk)
+                            chunk = FS.read(chunk_size)
+                    hashsum = file_hash.hexdigest()
```

Comment:
question: what is the case when event file has been changed after tf logs are converted?

Reply:
I think we can keep things simple here. Just keep the most recent conversion time (per run) and process files modified after that.

Reply:
fixed duplicate values in 66c5229

Reply:
Just figured out the kernel limitation only applies to the creation time so we're good to stick with modification time, will replace with hashsum generation part